questions/src/main/java/com/ds/questions/yashi/AddDigitByDigit.java

```diff
@@ -0,0 +1,57 @@
+package com.ds.questions.yashi;
+
+import java.util.Arrays;
+
+/*Given two input array of size X >= 0 or Y >= 0 (Can be of same size also!)
+ * A = [2, 4, 5, 1]
+ * B = [1, 7, 9]
+ *
+ * Perform addition of both the arrays digit by digit, such as:-
+ *
+ *       2 4 5 1
+ *         1 7 9
+ *      _________
+ *       2 6 3 0
+ *
+ * Return new array of result Result: [2, 6, 3, 0]!
+ * */
+public class AddDigitByDigit {
+    private static final int[] arrayA = {2, 4, 9};
+    private static final int[] arrayB = {1, 7, 9};
+
+    public static void main(String[] args) {
+        int[] result = preformAdditionDigitByDigit(arrayA, arrayB);
+        System.out.println(Arrays.toString(result));
+    }
+
+
+    public static int[] preformAdditionDigitByDigit(int[] arrayA, int[] arrayB) {
+
+        int out[];
+        if (arrayA.length == arrayB.length) {
+            out = new int[arrayA.length + 1];
+        } else if (arrayA.length > arrayB.length) {
+            out = new int[arrayA.length + 1];
+        } else {
+            out = new int[arrayB.length + 1];
+        }
```

Comment:
You are creating an array with extra element. Which is not required every time. For input `arrayA = {2, 4, 9} and arrayB = {1, 7, 9}` Your output is correct but adding extra zero at beginning. `[0, 4, 2, 8]`

Reply:
Thanks for your comment. Yes, I have added extra length just because of carry which comes then we need to handle that. Because the array is not dynamic that is why I have taken extra length.